From 8a0267e38cd4272dda00870f5f4699732aa1bb1e Mon Sep 17 00:00:00 2001
From: Andy Duan <fugang.duan@nxp.com>
Date: Wed, 22 Mar 2017 16:40:09 +0800
Subject: [PATCH 1607/5242] MLK-14498-13 tty: serial: imx: use
 tty_port_suspended() instead of ASYNC_SUSPENDED

commit  a867bd858575c34a7796e49fd8b5eb2e4b267b99 from
https://source.codeaurora.org/external/imx/linux-imx.git

tty_port flag "ASYNC_SUSPENDED" has been discarded from kernel upgrade, then
use .tty_port_suspended() instead of the flag check.

Signed-off-by: Fugang Duan <fugang.duan@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/tty/serial/imx.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/tty/serial/imx.c b/drivers/tty/serial/imx.c
index 6229462..1a0c147 100644
--- a/drivers/tty/serial/imx.c
+++ b/drivers/tty/serial/imx.c
@@ -1373,7 +1373,7 @@ static int imx_uart_startup(struct uart_port *port)
 	u32 ucr1, ucr2, ucr4;
 
 	/* some modem may need reset */
-	if (!(tty_port->flags & ASYNC_SUSPENDED)) {
+	if (!tty_port_suspended(tty_port)) {
 		retval = device_reset(sport->port.dev);
 		if (retval && retval != -ENOENT)
 			return retval;
-- 
1.7.9.5

